app/controllers/work_packages/auto_completes_controller.rb

```diff
@@ -37,11 +37,23 @@ def index
     q = params[:q].to_s
 
     if q.present?
-      query = (params[:scope] == 'all' && Setting.cross_project_work_package_relations?) ? WorkPackage : @project.work_packages
+      if params[:scope] == 'relatable'
+        unless @project
+          render_404
+          return
+        end
 
-      @work_packages |= query.visible.find_all_by_id(q.to_i) if q =~ /\A\d+\z/
+        scope = Setting.cross_project_work_package_relations? ? WorkPackage : @project.work_packages
+      elsif @project.nil? || params[:scope] == 'all'
+        scope = WorkPackage
+      else
+        scope = @project.work_packages
+      end
 
-      @work_packages |= query.visible.find(:all,
+      # query for exact ID matches first, to make an exact match the first result of autocompletion
+      @work_packages |= scope.visible.find_all_by_id(q.to_i) if q =~ /\A\d+\z/
+
+      @work_packages |= scope.visible.find(:all,
```

Comment:
@NobodysNightmare not sure when this (or #2104 will get merged in), but might also be a good opportunity to get rid of the `find(:all, ..` with conditions hash (deprecated in Rails 4.0, removed in Rails 4.1).